plugins/cilium-cni/cmd/interface.go

```diff
@@ -15,38 +15,55 @@ import (
 )
 
 func interfaceAdd(ipConfig *current.IPConfig, ipam *models.IPAMAddressResponse, conf *models.DaemonConfigurationStatus) error {
+	if ipam == nil {
+		return fmt.Errorf("missing IPAM configuration")
+	}
 	// If the gateway IP is not available, it is already set up
 	if ipam.Gateway == "" {
 		return nil
 	}
 
-	var masq bool
-	if ipConfig.Address.IP.To4() != nil {
-		masq = conf.MasqueradeProtocols.IPV4
-	} else {
-		masq = conf.MasqueradeProtocols.IPV6
-	}
+	var allCIDRs []*net.IPNet
 
-	allCIDRs := make([]*net.IPNet, 0, len(ipam.Cidrs))
 	for _, cidrString := range ipam.Cidrs {
 		_, cidr, err := net.ParseCIDR(cidrString)
 		if err != nil {
 			return fmt.Errorf("invalid CIDR '%s': %w", cidrString, err)
 		}
 		allCIDRs = append(allCIDRs, cidr)
 	}
+
 	// Coalesce CIDRs into minimum set needed for route rules
 	// The routes set up here will be cleaned up by linuxrouting.Delete.
 	// Therefor the code here should be kept in sync with the deletion code.
-	ipv4CIDRs, _ := ip.CoalesceCIDRs(allCIDRs)
-	cidrs := make([]string, 0, len(ipv4CIDRs))
-	for _, cidr := range ipv4CIDRs {
-		cidrs = append(cidrs, cidr.String())
+	ipv4CIDRs, ipv6CIDRs := ip.CoalesceCIDRs(allCIDRs)
+	coalescedCIDRs := make([]string, 0, len(allCIDRs))
+	var masq bool
+
+	if ipConfig.Address.IP.To4() != nil {
+		for _, cidr := range ipv4CIDRs {
+			coalescedCIDRs = append(coalescedCIDRs, cidr.String())
+		}
+
+		masq = conf.MasqueradeProtocols.IPV4
+	} else {
+		for _, cidr := range ipv6CIDRs {
+			coalescedCIDRs = append(coalescedCIDRs, cidr.String())
+		}
+
+		masq = conf.MasqueradeProtocols.IPV6
+	}
+	var ipamGateway string
+
+	if ipam != nil {
+		ipamGateway = ipam.Gateway
+	} else {
+		ipamGateway = ""
```

Comment:
This statement here is unreachable. We already check `ipam != nil` as the first thing in `interfaceAdd`. You can just keep using `ipam.Gateyway`, since here it's already guaranteed thanks to previous if checks that `ipam` is not nil and `ipam.Gateway` is not empty.